data/src/apiKey/java/org/cryptomator/data/cloud/pcloud/PCloudImpl.kt

```diff
@@ -286,13 +285,73 @@ internal class PCloudImpl(context: Context, cloud: PCloud) {
 		}
 	}
 
+	private fun readFile(filePath : String, sink: DataSink, listener : ProgressListener) {
+		var attempts = 0
+		while (++attempts <= MaxContentLinkDownloadAttempts) {
+			val fileLink = apiClient.createFileLink(filePath, DownloadOptions.DEFAULT).execute()
+			try {
+				// Attempt to download the link's content, starting with the best link variant.
+				for (url in fileLink.urls()) {
+					try {
+						fileLink.download(url, sink, listener)
+					} catch (e : APIHttpException) {
+						// HTTP 404's denote that the file may have been moved on another
+						// storage service node.
+						if (e.code == 404) {
+							// Check if more link variants are available, either try opening
+							// the next variant or give up by fall-through and throwing.
+							if (url != fileLink.urls().last()) {
+								continue
+							}
+						}
+
+						throw e
+					}
+				}
```

Comment:
```suggestion
				}
				break
```

We need to add a break directly after the `for`-loop, otherwise the `while`-loop will be executed for `MaxContentLinkDownloadAttempts`, regardless of whether there was an error or not. If it was successful it should be executed only once.

Reply:
@SailReal , see my latest changes, the unnecessary loops on success shouldn't be a problem anymore.